src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/Rbac/CosmosRole.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.ControlPlane.Core.Features.Rbac.Roles;
+using Microsoft.Health.CosmosDb.Features.Storage;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.ControlPlane.CosmosDb.Features.Storage.Rbac
+{
+    public class CosmosRole : Role
+    {
+        internal const string RolePartition = "Role_";
+
+        public CosmosRole(Role role)
+              : base(role.Name, role.ResourcePermissions,  role.Version)
+        {
+            ETag = $"\"{role.Version}\"";
+        }
+
+        [JsonConstructor]
+        protected CosmosRole()
+        {
+        }
+
+        [JsonProperty("id")]
+        public string Id => Name;
+
+        [JsonProperty("partitionKey")]
+        public string PartitionKey { get; } = RolePartition;
+
+        [JsonProperty("_etag")]
```

Comment:
There's a `KnownDocumentProperties.Etag` now. Also, there are known ones for `PartitionKey` and `Id`